examples/bots/discord_bot.py

```diff
@@ -0,0 +1,120 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import asyncio
+from typing import TYPE_CHECKING, Optional
+
+from camel.bots import DiscordApp
+from examples.bots.agent import Agent
+
+if TYPE_CHECKING:
+    from discord import Message
+
+
+class DiscordBot(DiscordApp):
+    r"""A Discord bot that listens for messages, adds them to a queue,
+    and processes them asynchronously.
+
+    This class extends the functionality of `DiscordApp` and adds message
+    handling by pushing messages into a queue for further processing.
+
+    Args:
+        msg_queue (asyncio.Queue): A queue used to store incoming messages for
```

Comment:
why we didn't add this to `DiscordApp `?

Reply:
Because `DiscordApp` is designed to focus on obtaining data from the Discord platform. When we can abstract a unified App, I think we can consider putting queue into it as an option.